sdv/metadata/single_table.py

```diff
@@ -410,12 +411,19 @@ def _detect_columns(self, data):
 
             if sdtype in self._REFERENCE_TO_SDTYPE.values() or sdtype == 'unknown':
                 column_dict['pii'] = True
+                if first_pii_field is None:
+                    first_pii_field = field
+
             elif sdtype == 'datetime' and dtype == 'O':
                 datetime_format = get_datetime_format(column_data.iloc[:100])
                 column_dict['datetime_format'] = datetime_format
 
             self.columns[field] = deepcopy(column_dict)
 
+        # When no primary key column was set, choose the first pii field
+        if self.primary_key is None and first_pii_field:
+            self.primary_key = first_pii_field
```

Comment:
Just to clarify, is the logic we want here to set the primary key to the first ID or PII column we find? Or is it to fall back on PII columns only if we don't find an ID column in the table?

Reply:
The issue doesn't specify, so I implemented the second approach. I could change this if we have evidence that PII columns are just as likely as ID ones to be selected as primary keys.